lib/rules/selector-max-id/__tests__/index.js

```diff
@@ -198,6 +218,52 @@ testRule(rule, {
       message: messages.expected("#foo:hover > #bar #baz", 2),
       line: 1,
       column: 8
+    },
+    {
+      code: "#foo #bar #baz { @media print { #bar {} } }",
+      description:
+        "compound selector: greater than max id selectors, nested @media query",
+      message: messages.expected("#foo #bar #baz", 2),
+      line: 1,
+      column: 1
+    },
+    {
+      code: "#foo #bar #baz { @if ($p == 1) { #bar {} } }",
+      description:
+        "compound selector: greater than max id selectors, nested @if statement",
+      message: messages.expected("#foo #bar #baz", 2),
+      line: 1,
+      column: 1
+    },
+    {
+      code: ".foo { @if ($p == 1) { #bar #baz #foo {} } }",
+      description:
+        "nested compound selector: greater than max id selectors, nested @if statement",
+      message: messages.expected(".foo #bar #baz #foo", 2),
+      line: 1,
+      column: 24
+    },
```

Comment:
I think tests containing `@if` should go to `test` function with `syntax: "scss"`.